### DISCUSSION TITLE
Async into var

### DISCUSSION THREAD

**Discussion by Frostswing** [react-native-async-storage/async-storage#890] [Q&A] [answered]
This is a very silly question, but for some reason i have found no answer to the past 3 days...

All i want to do is to put data from async storage and put it into a var...

For example my app could store the user's requestend language and reuse this data on the next app launch...

Using simple:
`value= AsyncStorage.getItem('chosenLanguage')`
wouldn't work since i get the promise instead of the data, this part i understand. 

After a short check i saw that since its an async function i should use await like this:
`var value=await AsyncStorage.getItem('@APP:chosenLanguage') ;` 
But then i get :
>Unexpected identifier '_asyncStorage'. Expected ')' to end an argument list. 
ERROR

I tried using:
```
 const getData = async (callback) => {
  var value,collect;
   try {
     value= await AsyncStorage.getItem('chosenLanguage').then(
       (avalue) => {
          collect=avalue;
       });
  
     } catch(e) {
     // error reading value
   }
   callback(collect) ;
 }
```
But it seems that while i can read collect and use it inside the getData function, i cannot use this data outside of the function...
Using a return to the avalue\collect variables gets me the promise instead of the data....

Can anyone help me please?
Thanks!

**Comment by Frostswing** [comment]
anyone has a solution?

**Reply by ansh** [reply]
Use Jotai which integrates with AsyncStorage

**Reply by Frostswing** [reply]
So to use AsyncStorage i need another library?

What is the main usage of this library then?
To be integrated inside usestates?

**Comment by ansh** [comment, marked as answer]
No, you can use a useEffect and also make it happen. AsyncStorage, is, like the name says, async.

You want a sync solution. Jotai is the easiest one. You can also set it up using useEffect. The solution should be trivial so I won’t include the code here.

**Reply by Frostswing** [reply]
tbh i just wanted local storage for startup values, like chosen app language ect... 
I guess ill try this or the sync-storage library on the community.

Thank you!

**Comment by ansh** [comment]
You can do that within a useEffect, just wait for the loading to finish below hiding the SplashScreen.
